cmd/handler-utils.go

```diff
@@ -116,16 +116,17 @@ func extractMetadataFromHeader(header http.Header) map[string]string {
 	}
 	// Go through all other headers for any additional headers that needs to be saved.
 	for key := range header {
-		cKey := http.CanonicalHeaderKey(key)
-		if strings.HasPrefix(cKey, "X-Amz-Meta-") {
-			metadata[cKey] = header.Get(key)
-		} else if strings.HasPrefix(key, "X-Minio-Meta-") {
-			metadata[cKey] = header.Get(key)
+		if key != http.CanonicalHeaderKey(key) {
```

Comment:
I don't understand why this is needed - the only way to get a non-canonical header is if we add code that changes request headers. Request headers are not supposed to be modified - any code that does so is introducing bugs. So checking for this and returning errors to the user does not appear to be correct, as they could not have caused such a problem in the first place.